src/version.ts

```diff
@@ -0,0 +1,179 @@
+import chalk from 'chalk'
+import fs from 'fs'
+import FileHelpers from './filehelpers'
+// eslint-disable-next-line
+const minimist = require("minimist");
+
+const args = minimist(process.argv.slice(2))
+
+/**
+ * MiraVersion checks the CDK version consistency across the Mira packge and the
+ * applications using Mira.  It is capable of force-patching package.json files
+ * and running the NPM install.
+ */
+export class MiraVersion {
+    static instance: MiraVersion
+    constructor () {
+      if (!MiraVersion.instance) {
+        MiraVersion.instance = this
+      }
+    }
+
+    /**
+     * Checks a particular dependency CDK version.
+     */
+    /* eslint-disable-next-line */
+    checkApplicationDependencyCDKVersion (pkg: any, dep: string, autoFix = true): boolean {
+      if (!dep.startsWith('@aws-cdk') && !dep.startsWith('aws-cdk')) {
+        return false
+      }
+      const cdkVersion = pkg.dependencies[dep]
+      const miraVersion = this.getLocalMiraCDKVersion()
+      if (typeof miraVersion !== 'string') {
+        throw new Error('Mira contains CDK version issues.')
+      }
+      try {
+        this.checkDependency(cdkVersion, dep, miraVersion, 'Mira')
+      } catch (e) {
+        console.info(`${chalk.cyan('Info:')} Found a version mismatch between Mira and ${pkg.name}.` +
+                `\n\t${pkg.name} is ${cdkVersion}` +
+                `\n\tMira is ${miraVersion}`)
+        if (autoFix) {
+          pkg.dependencies[dep] = miraVersion
+          return true
+        }
+      }
+      return false
+    }
+
+    /**
+     * Gets the version of CDK used by an application.  If there is a version
+     * mismatch, then modifies the package.json file of the application, informs
+     * the user and exits.
+     */
+    checkApplicationCDKVersion (autoFix = true): void {
+      const pkgFile = FileHelpers.getPackageDirectory() + '/package.json'
+      const pkg = JSON.parse(fs.readFileSync(pkgFile, 'utf8'))
+
+      let madeChange = false
+      for (const dep in pkg.dependencies) {
```

Comment:
Shouldn't we need to check the `devDependencies` as well?

Reply:
Addressed in latest commit